### PR TITLE
Add method 'addStructInfo' in DictMethodCompModelHelper

### PR DESCRIPTION
**Pull request by brindakv**
This PR addresses RO-3238 to populate missing `_struct.title` for computed structure models.

### CODE REVIEW

File: rcsb/utils/dictionary/DictMethodCompModelHelper.py [hunk 1]
```diff
@@ -340,3 +341,56 @@ def buildCompModelProvenance(self, dataContainer, catName, **kwargs):
...
+            bool: True for success or False otherwise
+
+        """
+        logger.debug("Starting with %r %r %r", dataContainer.getName(), catName, kwargs)
+        try:
+            # Check if struct table is already present
+            if dataContainer.exists("struct"):
+                return False
+            # Check for all categories/data items required for the ETL (will return False for experimental models)
+            if not (dataContainer.exists("entry") and dataContainer.exists("entity") and dataContainer.exists("ma_data")):
```
Comment (piehld) [inline rcsb/utils/dictionary/DictMethodCompModelHelper.py:362]: Won't this cause the function to exit early for all computed-models?

File: rcsb/utils/dictionary/DictMethodCompModelHelper.py [hunk 2]
```diff
@@ -340,3 +341,56 @@ def buildCompModelProvenance(self, dataContainer, catName, **kwargs):
...
+            entryId = tObj.getValue("id", 0)
+
+            methodType = "computational"
+            tL = []
+            qObj = dataContainer.getObj("entity")
+
+            for ii in range(qObj.getRowCount()):
+                pdbxDesc = qObj.getValue("pdbx_description", ii)
+                eType = qObj.getValue("type", ii)
+                if pdbxDesc not in [".", "?"] and eType in ["polymer", "Polymer", "POLYMER"]:
```
Comment (piehld) [inline rcsb/utils/dictionary/DictMethodCompModelHelper.py:380]: I would use this instead:
```python
                if pdbxDesc not in [".", "?"] and eType.lower() == "polymer":
```

**Review by piehld — COMMENTED** [review]
Looks good, thanks! But had some comments/questions. Also, I'm guessing you'll add this method to the dictionary files too?

The other thing that might be good to add is a note in the code somewhere to remind us to remove this method once the incoming structures all have `_struct` already (i.e., make it clear that this method is supposed to just be a temporary workaround).

File: rcsb/utils/dictionary/DictMethodCompModelHelper.py (hunk 1 above)
Comment (piehld) [reply rcsb/utils/dictionary/DictMethodCompModelHelper.py:362]: Whoops, I overlooked the `not`. Never mind :).

**Comment by brindakv** [comment]
> Looks good, thanks! But had some comments/questions. Also, I'm guessing you'll add this method to the dictionary files too?
> 
Yes, that's already updated. 

> The other thing that might be good to add is a note in the code somewhere to remind us to remove this method once the incoming structures all have `_struct` already (i.e., make it clear that this method is supposed to just be a temporary workaround).

The method will do nothing if `struct` is already populated (see lines 359 and 360). It will make a difference only for computed structure models that do not have the `struct` table. Considering that these models are from external resources, `struct` is not mandatory, and our pipeline breaks if `struct` is missing, I think it is reasonable to leave the method as is to handle any stray cases.

**Comment by brindakv** [comment]
> OK, thanks for the clarification. I agree that it's good to keep the method in here to handle any future stray cases, but I still feel like it's a good idea to make a note on why we created the method in the first place (i.e., to handle incomplete AF models). That way, if/when we eventually run into a case where the title is being populated in an undesirable way, we know why the method is the way it is. That's just my strong preference, but you're welcome to disagree.

Thanks. There is already a comment that the method adds missing `struct` table.  I'd rather leave it as is.

**Comment by piehld** [comment]
OK, very well :)

**Pull request merged**